### PR TITLE
ransack search for question

### PR DESCRIPTION
**Pull request by nguyenngocha**

### CODE REVIEW

File: app/controllers/admins/questions_controller.rb [hunk 1]
```diff
@@ -1,18 +1,28 @@
 class Admins::QuestionsController < ApplicationController
-  load_and_authorize_resource
-  before_action :load_sources, only: [:new, :edit]
-  before_action :load_subject_form, only: [:edit, :update]
+  before_action :authenticate_admin!
+  load_and_authorize_resource except: :create
```
Comment (kagoromo) [inline app/controllers/admins/questions_controller.rb]: skip_load_resource only: :create

**Comment by kagoromo** [comment]
OK

**Comment by ngtrduc** [comment]
ok

File: app/controllers/admins/questions_controller.rb [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
+  before_action :authenticate_admin!
   load_and_authorize_resource
-  before_action :load_sources, only: [:new, :edit]
-  before_action :load_subject_form, only: [:edit, :update]
+  skip_load_resource only: :create
+  before_action :load_sources, only: [:new, :edit, :index]
+  before_action :load_question_form, only: [:edit, :update]
+
+  def index
+    @search = @questions.ransack(params[:q])
```
Comment (ngtrduc) [inline app/controllers/admins/questions_controller.rb]: bo ()

File: app/controllers/users/questions_controller.rb [hunk 3]
```diff
@@ -10,8 +11,12 @@ def new
   end
 
   def index
-    @questions = current_user.questions.page(params[:page]).
+    @questions = current_user.questions
+    @search = @questions.ransack(params[:q])
```
Comment (ngtrduc) [inline app/controllers/users/questions_controller.rb]: bo ()

File: app/controllers/admins/questions_controller.rb [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
+  skip_load_resource only: :create
+  before_action :load_sources, only: [:new, :edit, :index]
+  before_action :load_question_form, only: [:edit, :update]
+
+  def index
+    @search = @questions.ransack params[:q]
+    @questions = @search.result unless params[:q].nil?
+    @questions = @questions.page(params[:page]).
+      per Settings.admin.suggested_question.per_page
+    @status = Question.statuses
```
Comment (hellovietnam93) [inline app/controllers/admins/questions_controller.rb]: so ít hay số nhiều ?

**Pull request merged**